OpenRA.Mods.Common/Traits/Buildings/Gate.cs

```diff
@@ -111,13 +112,7 @@ void INotifyBlockingMove.OnNotifyBlockingMove(Actor self, Actor blocking)
 
 		bool CanRemoveBlockage(Actor self, Actor blocking)
 		{
-			return !self.IsDisabled() && BuildComplete && blocking.AppearsFriendlyTo(self);
-		}
-
-		protected override void AddedToWorld(Actor self)
-		{
-			base.AddedToWorld(self);
-			blockedPositions = Info.Tiles(self.Location);
```

Comment:
`blockedPositions` still needs to be updated if the actor is manually removed and readded to the world (e.g. using Lua).